frontend/src/component/feature/FeatureView/FeatureOverview/ReleasePlan/SafeguardForm/SafeguardForm.tsx

```diff
@@ -508,11 +543,11 @@ const SafeguardFormBase: FC<SafeguardFormBaseProps> = ({
                                     inputProps={{
                                         step: 0.1,
                                     }}
-                                    value={threshold}
-                                    onChange={(e) => {
-                                        const value = e.target.value;
-                                        handleThresholdChange(Number(value));
-                                    }}
+                                    value={thresholdInputValue}
```

Comment:
if this component is closely related with the custom hook above we can extract them together and have a component + its own hooks living together